peer/brontide_test.go

```diff
@@ -33,223 +29,209 @@ var (
 	)
-	if err != nil {
-		t.Fatalf("error creating close proposal: %v", err)
-	}
+	require.NoError(t, err)
 
 	parsedSig, err := lnwire.NewSigFromSignature(bobSig)
-	if err != nil {
-		t.Fatalf("error parsing signature: %v", err)
-	}
-	closingSigned := lnwire.NewClosingSigned(chanID, aliceFee, parsedSig)
-	alicePeer.chanCloseMsgs <- &closeMsg{
-		cid: chanID,
-		msg: closingSigned,
-	}
+	require.NoError(t, err)
+
+	bobClosingSigned := lnwire.NewClosingSigned(
+		chanID, aliceFee, parsedSig,
+	)
+	err = pushMessage(aliceConn, bobClosingSigned)
+	require.NoError(t, err)
 
-	// Alice should now see that we agreed on the fee, and should broadcast the
-	// closing transaction.
+	// Alice should now see that we agreed on the fee, and should broadcast
+	// the closing transaction.
 	select {
-	case <-broadcastTxChan:
+	case <-ctx.publTx:
 	case <-time.After(timeout):
 		t.Fatalf("closing tx not broadcast")
 	}
 
-	// Need to pull the remaining message off of Alice's outgoing queue.
-	select {
-	case outMsg := <-alicePeer.outgoingQueue:
-		msg = outMsg.msg
-	case <-time.After(timeout):
-		t.Fatalf("did not receive ClosingSigned message")
-	}
-	if _, ok := msg.(*lnwire.ClosingSigned); !ok {
-		t.Fatalf("expected ClosingSigned message, got %T", msg)
-	}
+	// Alice will respond with a ClosingSigned.
+	msg, err = getMessage(aliceConn)
+	require.NoError(t, err)
+	aliceClosingSigned, ok = msg.(*lnwire.ClosingSigned)
+	require.True(t, ok)
+	require.Equal(t, aliceClosingSigned.FeeSatoshis, aliceFee)
 
 	// Alice should be waiting in a goroutine for a confirmation.
-	notifier.ConfChan <- &chainntnfs.TxConfirmation{}
+	ctx.notifier.ConfChan <- &chainntnfs.TxConfirmation{}
 }
 
 // TestPeerChannelClosureAcceptFeeInitiator tests the shutdown initiator's
 // behavior if we can agree on the fee immediately.
 func TestPeerChannelClosureAcceptFeeInitiator(t *testing.T) {
 	t.Parallel()
 
-	notifier := &mock.ChainNotifier{
-		SpendChan: make(chan *chainntnfs.SpendDetail),
-		EpochChan: make(chan *chainntnfs.BlockEpoch),
-		ConfChan:  make(chan *chainntnfs.TxConfirmation),
-	}
-	broadcastTxChan := make(chan *wire.MsgTx)
+	ctx := createTestContext(t)
+	defer ctx.cleanup()
 
-	alicePeer, bobChan, cleanUp, err := createTestPeer(
-		notifier, broadcastTxChan, noUpdate,
-	)
-	if err != nil {
-		t.Fatalf("unable to create test channels: %v", err)
-	}
-	defer cleanUp()
+	aliceConn := ctx.aliceConn
+
+	// Send the Init message to Alice.
+	err := pushMessage(aliceConn, ctx.bobInit)
+	require.NoError(t, err)
+
+	// Alice's Peer struct is created, representing a connection to Bob.
+	alicePeer := NewBrontide(*ctx.aliceCfg)
+	require.NoError(t, alicePeer.Start())
+
+	// Cleanup Alice via Disconnect.
+	defer func() {
+		alicePeer.Disconnect(errDummy)
+		alicePeer.WaitForDisconnect(make(chan struct{}))
+	}()
+
+	// Alice should reply with an Init message, we discard it.
+	msg, err := getMessage(aliceConn)
+	require.NoError(t, err)
+	_, ok := msg.(*lnwire.Init)
```

Comment:
nit: instead of these casts, we could use `reqire.IsType` to make the tests more readable.

Reply:
Changed only for the `Init` messages, other message types are cast and then the fields are used